$\require{cancel}  \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols} \require{sinuitx} \require{mhchem}$
<div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom"><a href="https://www.mathjax.org">
    <img style="height:1.25em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" />
</a></span> </div>

<header style="background-color:#e7e2ee; color:#461d7c; line-height:2em; padding-left:1em; padding-right:1em; bottom:0; float:bottom; text-align:right; float:right; font-family:serif; font-size:small;">Problem 4-21 ⋮ Çengel, Y. A., and Boles, M. A. (2001). <i>Thermodynamics: An Engineering Approach.</i> Boston: McGraw-Hill.</header>  




### Given   
 1-kg of water that is initially at 90°C with a quality of 10 percent occupies a spring-loaded piston–cylinder device. This device is now heated until the pressure rises to 800 kPa and the temperature is 250°C. 


### Find 
Determine the total work produced during this process, in kJ. 

### Get Organized

Every good home has a good diagram.

![figure, cb8, 4-21](./figures/ch04_cen98179_p04021_lg.jpg)

These things you should notice right away:
1. Closed system ⋮ constant mass.  
1. We can resolve both states ⋮ $(T_1,x_1),\ (P_2,T_2),$ all given.
1. Boundary work is the only form of work stated.
1. We have to account for a spring $\left(\Delta F_{\rm spring} = k \Delta x\right).$ 

We have two states we can resolve, but are not told the path the system takes to go from one to the other. All we need is a little critical thinking:

Our system pressure is dictated by contributions from the following:  
1. Atmospheric pressure $P_{\rm atm}$, which is constant;   
1. Piston weight $\left(\Delta P_{\rm piston} = m_{\rm piston}g/P_{\rm piston} \right),$ also constant; and   
1. Spring force $\left(\Delta F_{\rm spring} = k \Delta x\right)$, which is proportional to displacement. Considering constant area, it follows naturally that the spring's contribution to pressure is proportional to volume (ie, the relationship is linear: $\Delta P \propto \Delta V$.

### Attack

All of our work is boundary work.  We can resolve both states, and we know $\Delta P \propto \Delta V$. Hopefully, you realize this means our process forms a trapezoid on a PV diagram. 

![PV diagram](figures/cb8,4-021,PVdiagram.png)

\begin{gather*}
 W_{\rm b,out} = \int^2_1 P\,{\rm d}V = \frac{1}{2} \left( P_1 + P_2 \right)\left( V_2 - V_1 \right) \\[6pt]
 P_1 = P_\text{sat@90°C} \quad
 P_2 = 800\,{\rm kPa} \quad
 V_1 = mv_\text{90°C, x=10%} \quad 
 V_2 = mv_\text{250°C,800kPa}
\end{gather*}


In [8]:
using Unitful

In [9]:
m  = 1u"kg"                    # given
P1 = 0.7014u"bar"              # table
P2 = 800u"kPa"                 # given
x1 = 0.1                       # given

vf1 = 1.0360e-3u"m^3/kg"       # table
vg1 = 2.361u"m^3/kg"           # table
v1 = vf1 + x1 * (vg1 - vf1)    
v2 = 0.2932u"m^3/kg"           # table

W_out = m * (P1 + P2) * (v2 - v1) / 2 |> u"kJ"   # trapezoid

println("W_out = ", round(typeof(W_out),W_out,digits=3))

W_out

W_out = 24.437 kJ


24.436837731999994 kJ

<div class="btn-success">&nbsp;<div class="badge"></div><i class="fa fa-bomb"></i><strong> Boom</strong></div>
<div class="panel-success"><div class="panel-heading"><div class="badge"></div> 

    
\begin{equation*}
   W_{out} = 24.437\,\ce{kJ}
\end{equation*}  


</div></div>

### Let's plot it.

<div class="btn-primary">&nbsp;<div class="badge"></div><i class="fa fa-exclamation-circle"></i><strong> Info</strong><div>
<div class="panel-primary"><div class="panel-heading"><div class="badge"></div> 

The plot may not load automatically. If not, restart the session: go to the Jupyter menu $\rightarrow$ `Kernel` $\rightarrow$ `Restart & Run All`. It will take awhile to run everything, and you may see warnings, but it will eventually give you an interactive plot. 

![jupyterRerun.png](figures/jupyterRerun.png)

</div></div>

In [10]:
using Plots, CoolProp
plotlyjs()

Plots.PlotlyJSBackend()

In [11]:
"""
Prep the process numbers: consistent units, etc.
It's overkill for 2 points, but these steps are good,
no matter how many states we have.
"""
vs = [v1, v2] .|> u"m^3/kg" 
Ps = [P1, P2] .|> u"Pa" 

process = [vs Ps] .|> ustrip;

In [12]:
#___ make an array of water Pv saturation dome  

P=range(5000, stop=PropsSI("Pcrit","water"), length=500) #|> collect

Pvfg = reshape(
        [P 
        (PropsSI.("D","P",P,"Q",0,"water")).^(-1) 
        (PropsSI.("D","P",P,"Q",1,"water")).^(-1)],
        500,3);

In [13]:
#___ plot the dome and our process from the problem ______

plot(Pvfg[:,2:3], Pvfg[:,1], label = ["sat liquid" "sat vapor"])
xaxis!("sp volume [m³/kg]", :log10)
yaxis!("pressure [Pa]")

scatter!(process[:,1],process[:,2], label = "process", fillrange=0, fillalpha=0.8)

<!DOCTYPE html>
 
 
 Plots.jl